### PR TITLE
chore: extend README to include up-to-date examples and a high level overview of available functions.

### PR DESCRIPTION
**Pull request by tylerburdsall**
Rendered markdown: https://github.com/kvc0/goodmetrics_rs/blob/038ba21e1baee82fcb5f8de3cfe315eb8ed61b93/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,100 @@
...
+
+* Useful for identifying percentiles of data, like the 99th percentile of request sizes.
+* Generates a `Histogram` or `ExponentialHistogram` depending on configuration.
+
+#### `metrics.sum("name", value)`
+
+* Useful for tracking a simple counter such as bytes stored on disk, or number of tokens consumed.
+* Generates a `Gauge`
+
+#### `metrics.time("name")`
```
Comment (kvc0) [inline README.md:93]:
```suggestion
#### `let _guard = metrics.time("name")`
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,100 @@
...
+
+#### `metrics.distribution("name", value)`
+
+* Useful for identifying percentiles of data, like the 99th percentile of request sizes.
+* Generates a `Histogram` or `ExponentialHistogram` depending on configuration.
+
+#### `metrics.sum("name", value)`
+
+* Useful for tracking a simple counter such as bytes stored on disk, or number of tokens consumed.
+* Generates a `Gauge`
```
Comment (kvc0) [inline README.md:91]:
```suggestion
Adds up all the sum calls for "name" within each reporting period.

* Useful for tracking a counter such as bytes stored on disk, or number of tokens consumed.
* Aggregates locally into a `Sum`.
```

File: README.md [hunk 3]
```diff
@@ -1,24 +1,100 @@
...
+
+#### `metrics.time("name")`
+
+* For measuring how long something takes within the given scope of code.
+* Generates a `Histogram` or `ExponentialHistogram` of latency values in nanoseconds, depending on configuration.
```
Comment (kvc0) [inline README.md:96]:
```suggestion
* Aggregates locally into `Histogram` or `ExponentialHistogram`, depending on configuration.
* Unit for timers is nanoseconds.
```

File: README.md [hunk 4]
```diff
@@ -1,24 +1,100 @@
...
+
+#### `metrics.measurement("name", value)`
+
+* Use when your aggregation of data requires things like the `min`, `max`, `sum`, and `count` of data points received.
+* Generates a `StatisticsSet`
+
+#### `metrics.distribution("name", value)`
+
+* Useful for identifying percentiles of data, like the 99th percentile of request sizes.
+* Generates a `Histogram` or `ExponentialHistogram` depending on configuration.
```
Comment (kvc0) [inline README.md:86]:
```suggestion
* Aggregates locally into a `Histogram` or `ExponentialHistogram` depending on configuration.
```

File: README.md [hunk 5]
```diff
@@ -1,24 +1,100 @@
...
+### Available metrics functions and when to use them
+
+Latest docs can always be found on [docs.rs](https://docs.rs/goodmetrics/latest/goodmetrics/).
+
+A basic knowledge of [OpenTelemetry metric types](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#metric-points) is expected, but a shortened guide can be summarized below:
+
+#### `metrics.measurement("name", value)`
+
+* Use when your aggregation of data requires things like the `min`, `max`, `sum`, and `count` of data points received.
+* Generates a `StatisticsSet`
```
Comment (kvc0) [inline README.md:81]:
```suggestion
* Use when you only need `min`, `max`, `sum`, and `count` of data points received.
* Use `sum()` instead if you only need `sum` or `count`.
* Use `distribution()` instead if you need percentiles of values.
* Aggregates locally into a `StatisticsSet`
```

File: README.md [hunk 6]
```diff
@@ -1,24 +1,100 @@
...
+        metrics.sum("errors", 1_u64)
+    }
+}
+```
+
+### Available metrics functions and when to use them
+
+Latest docs can always be found on [docs.rs](https://docs.rs/goodmetrics/latest/goodmetrics/).
+
+A basic knowledge of [OpenTelemetry metric types](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#metric-points) is expected, but a shortened guide can be summarized below:
```
Comment (kvc0) [inline README.md:76]: I think if you want to put this in here it should go in a separate heading below this, `### Opentelemetry translation` or something like that. The translation per-type is kind of interesting, and probably deserves some explanation!

**Review by kvc0 — COMMENTED** [review]
Thank you for this documentation update! I'd be happy to chat with you separately about any of my feedback, and of course I'm open to your perspective on the things I'm pointing out.

File: README.md (hunk 6 above)
Comment (tylerburdsall) [reply README.md:76]: Upon further reflection. we can always omit this

Comment (tylerburdsall) [reply README.md]: https://github.com/kvc0/goodmetrics_rs/pull/19/commits/aadb97db80102ea54997e25e0db4f983fd1ec7b0

**Review by kvc0 — APPROVED** [review]
This looks great, thank you!

**Pull request merged**